duckduckgo/macos-browser
# Improves NetworkProtectionDebugMenuController

Task/Issue URL: https://app.asana.com/0/0/1205135296966627/f
This is a follow up to: https://github.com/duckduckgo/macos-browser/pull/1378

## Stack:
- https://github.com/duckduckgo/macos-browser/pull/1378
   - https://github.com/duckduckgo/macos-browser/pull/1389 <- This PR
   - https://github.com/duckduckgo/macos-browser/pull/1397

## Description:

Implements some changes based on the feedback received in these two comments:
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1272187109
- https://github.com/duckduckgo/macos-browser/pull/1378#discussion_r1272244652

These changes were tackled separately for these reasons:
- The base PR was mostly about moving existing code to a different location.
- The base PR works fine.
- Reviewing these changes separately guarantees a faster and more focused feedback loop.

## Steps to test this PR:

1. Make sure the preferred server and rekey validity menus are populated correctly.
2. Make sure selecting an entry in these menus is properly updated in the menu UI.
3. Make sure selecting a different server results in the server being properly selected when NetP is connected.
4. Make sure selecting a different rekey validity period results in `NetworkDebugUtilities.swift:106` being called (use a breakpoint).

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
